Ex5/Ex5.java

```diff
@@ -0,0 +1,105 @@
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+public class Ex5 {
+    /**
+     * The function receives a dictionary (a lost of words in lexicographic order) of all words in an unknown language
+     * and find the alphabet (an ordered list off characters) of that language
+     *
+     * @param dictionary - a list of words
+     * @return - an ordered list of characters
+     */
+    public static List<Character> findAlphabetOrder(List<String> dictionary) {
+        Graph alphabetGraph = new Graph();
+        int index = 0;
+        List<String> updateDict = updateDictionary(dictionary, index);
+        List<List<String>> newDictionary = findSubDictionary(updateDict, index);
+
+        while (newDictionary != null) {
+            for (List<String> subDict : newDictionary) {
+                List<Character> subAlphabet = findAlphabetList(subDict, index);
+                alphabetGraph.updateGraph(subAlphabet);
+            }
+            index++;
+            updateDict = updateDictionary(updateDict, index);
+            newDictionary = findSubDictionary(updateDict, index);
+        }
+        return alphabetGraph.topologicalSort();
+    }
+
+    /**
+     * The function receives a dictionary (a list of words in lexicographic order) and an index, and return
+     * a list of dictionaries according to their prefixes
+     *
+     * @param dictionary - a list of words in lexicographic order
+     * @param index - the index of the char we check
+     * @return - a list of dictionaries
+     */
+    private static List<List<String>> findSubDictionary(List<String> dictionary, int index) {
+        List<List<String>> dictionaryList = new ArrayList<>();
+        List<String> subDictionary = new ArrayList<>();
+
+        for (int j = 0; j < dictionary.size(); j++) {

```

Comment:
When operating on `List`, be very careful with using indexes and `get(index)`! What is the complexity of `get(index)` in a `LinkedList`? How does that affect complexity of the whole method?

Safer approach would be to use `ListIterator` instead (condition `j == 0` can be replaced by `!iterator.hasPrevious()`).